### ISSUE TITLE
[BUG] UBSAN Error

### ISSUE THREAD

**Issue by jlsantiago0** [Haivision/srt#2830]
**Describe the bug**
Clang UBSAN indicates that the unaligned access of a char aligned buffer as an int and long is undefined behavior srt-1.5.3/srtcore/channel.h: lines 228-268 in various places.

**To Reproduce**

Built with Clang + UBSAN sanitizer. running as SRT listener.

**Expected behavior**
No undefined behavior.

**Screenshots**
Will attach sanitizer logs.

**Desktop (please provide the following information):**
 - OS: linux
 - SRT Version / commit ID: 1.5.3 release

**Additional context**
none

[ubsan.log](https://github.com/Haivision/srt/files/13590128/ubsan.log)

**Comment by ethouris** [comment]
I think it should suffice to declare alignment on `m_acCmsgSendBuffer` and the other, too. For this, some portable alignment declarator must be provided, like this:
```
#if HAVE_CXX11
#define SRT_ATR_ALIGNAS(n) alignas(n)
#elif HAVE_GCC
#define SRT_ATR_ALIGNAS(n) __attribute__((aligned(n))
#else
#define SRT_ATR_ALIGNAS(n)
#endif
```

and then use `SRT_ATR_ALIGNAS(8)` on the declaration of these buffers.

**Comment by maxsharabayko** [comment]
Resolved by #2844.
